plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -44,27 +44,28 @@ def __init__(self):
         executable_name = "CuraEngine"
         if Platform.isWindows():
             executable_name += ".exe"
-        default_engine_location = executable_name
+
+        engine_location = executable_name
         if os.path.exists(os.path.join(Application.getInstallPrefix(), "bin", executable_name)):
-            default_engine_location = os.path.join(Application.getInstallPrefix(), "bin", executable_name)
+            engine_location = os.path.join(Application.getInstallPrefix(), "bin", executable_name)
         if hasattr(sys, "frozen"):
-            default_engine_location = os.path.join(os.path.dirname(os.path.abspath(sys.executable)), executable_name)
-        if Platform.isLinux() and not default_engine_location:
+            engine_location = os.path.join(os.path.dirname(os.path.abspath(sys.executable)), executable_name)
+        if Platform.isLinux() and not os.path.exists(engine_location):
             if not os.getenv("PATH"):
                 raise OSError("There is something wrong with your Linux installation.")
             for pathdir in os.getenv("PATH").split(os.pathsep):
-                execpath = os.path.join(pathdir, executable_name)
+                execpath = os.path.join(pathdir, engine_location)
```

Comment:
Did you mean to change this one? It's not the renamed variable. This can have repercussions since the `engine_location` variable changes throughout multiple iterations of the for loop.